contracts/ChainmonstersRewards.cdc

```diff
@@ -259,6 +261,20 @@ pub contract ChainmonstersRewards: NonFungibleToken {
 
             return newID
         }
+        
+        // consuming an NFT (item) to be converted to in-game economy
+        pub fun consumeItem(token: @NonFungibleToken.NFT, playerId: String) {
+            let token <- token as! @ChainmonstersRewards.NFT
+
+            let id: UInt64 = token.id
+
+            // add the new token to the dictionary which removes the old one
+            let oldToken <- self.ownedNFTs[id] <- token
```

Comment:
This reassignment is not needed in this case. You can destroy the token variable directly